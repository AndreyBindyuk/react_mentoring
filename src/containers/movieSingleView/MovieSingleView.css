.movie-view-container {
  display: flex;
  flex-flow: row wrap;
  margin-left: 250px;
  width: 700px;
  height: 400px;
  justify-content: space-around;
  box-shadow: 0px 8px 20px 0px rgba(0, 0, 0, 0.15);
}

.movie-image {
  width: 40%;
  height: 80%;
  align-self: center;
  border-radius: 5px;
  overflow: hidden;
  margin-bottom: 20px;
  border: 1px solid transparent;
}

.movie-info-container {
  display: flex;
  flex-flow: column;
  align-self: center;
  width: 55%;
  height: 80%;
  box-shadow: 0px 8px 20px 0px rgba(0, 0, 0, 0.15);
}

.search {
  align-self: flex-end;
}

.movies-title {
  font-size: 20px;
  color: #84878d;
  font-weight: 300;
  margin-bottom: 10px;
}

.movie-meta {
  height: 30%;
  list-style: none;
  font-size: 15px;
  color: #84878d;
}

.movie-meta li {
  margin-bottom: 10px;
}
.movie-text {
  font-size: 15px;
  color: #84878d;
  margin-top: 5px;
  margin-bottom: 5px;
  align-self: center;
  height: 60%;
  width: 90%;
  box-shadow: 0px 0px 10px 5px rgba(0, 0, 0, 0.15);
}

.btn-s {
  min-width: 100px;
  height: 23px;
  padding: 0 15px;
  background: red;
  white-space: nowrap;
  border-radius: 0;
  font-size: 12px;
  color: #fff;
  transition: all 0.2s ease-out, color 0.2s ease-out;
  border: 0;
  cursor: pointer;
  font-weight: bold;
  box-shadow: 0px 2px 5px 0px rgba(0, 0, 0, 0.15);
}
