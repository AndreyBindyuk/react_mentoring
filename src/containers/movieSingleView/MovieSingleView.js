import React from "react";
import "./MovieSingleView.css";
import Rater from "react-rater";
import "react-rater/lib/react-rater.css";
import { fetchMovieItem } from "./fetchMovieItem";
import { connect } from "react-redux";

class MovieSingleView extends React.Component {
  constructor(props) {
    super(props);
  }

  componentDidMount() {
    this.props.fetchMovie();
  }

  render() {
    const {
      title,
      release_date,
      genres,
      poster_path,
      vote_count,
      overview
    } = this.props.movie;
    return (
      <div className="movie-view-container">
        <img className="movie-image" alt="#" src={poster_path} />
        <div className="movie-info-container">
          <button className="btn-s">SEARCH</button>
          <h1 className="movies-title">{title}</h1>
          <ul className="movie-meta">
            <li>
              <strong>Rating:</strong>
              <Rater total={10} rating={vote_count} interactive={false} />
            </li>
            <li>
              <strong>Длина:</strong> 180 min
            </li>
            <li>
              <strong>Премьера:</strong> {release_date}
            </li>
            <li>
              <strong>Категория:</strong> {genres}
            </li>
          </ul>
          <div className="movie-text">{overview}</div>
        </div>
      </div>
    );
  }
}

const mapDispatchToProps = {
  fetchMovie: fetchMovieItem
};

const mapStateToProps = state => ({
  movie: state.movieItem.movie
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(MovieSingleView);
