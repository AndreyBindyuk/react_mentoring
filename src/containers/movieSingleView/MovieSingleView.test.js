import React from "react";
import { shallow, mount } from "enzyme";
import MovieSingleView from "./MovieSingleView";

describe("<MovieSingleView />", () => {
  it("should render components correctly", () => {
    const component = shallow(<MovieSingleView />);
    expect(component).toMatchSnapshot();
  });
});
