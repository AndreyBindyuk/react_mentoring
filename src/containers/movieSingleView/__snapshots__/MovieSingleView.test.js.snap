// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieSingleView /> should render components correctly 1`] = `ShallowWrapper {}`;
